lightclient-circuits/src/committee_update_circuit.rs

```diff
@@ -311,25 +343,113 @@ mod tests {
     }
 
     #[test]
-    fn test_committee_update_circuit() {
+    fn test_prove_and_verify_plonk() {
         const K: u32 = 20;
+        let mut file = File::open("./params/kzg_bn254_20.srs").expect("Failed to open params!");
+
+        let mut buffer = Vec::new();
+
+        file.read_to_end(&mut buffer)
+            .expect("Failed to read params!");
+
+        let params: ParamsKZG<Bn256> =
+            ParamsKZG::read(&mut Cursor::new(&buffer)).expect("Failed to read buffer :(");
+
+        const PINNING_PATH: &str = "./config/committee_update_20.json";
+        const PKEY_PATH: &str = "../build/committee_update_20.pkey";
+
+        let pk = CommitteeUpdateCircuit::<Testnet, Fr>::create_pk(
+            &params,
+            PKEY_PATH,
+            PINNING_PATH,
+            &CommitteeUpdateArgs::<Testnet>::default(),
+            None,
+        );
+
+        println!("[Ok] Proving Key");
+
         let witness = load_circuit_args();
-        let params: ParamsKZG<Bn256> = gen_srs(K);
 
         let circuit = CommitteeUpdateCircuit::<Testnet, Fr>::create_circuit(
-            CircuitBuilderStage::Mock,
-            None,
+            CircuitBuilderStage::Prover,
+            Some(Eth2ConfigPinning::from_path(PINNING_PATH)),
             &witness,
             &params,
         )
-        .unwrap();
+        .expect("Failed to construct circuit");
+
+        println!("[Ok] Construct Circuit");
+
+        let reader: Vec<u8> = vec![];
+        let mut transcript_writer: Blake2bWrite<Vec<u8>, G1Affine, Challenge255<G1Affine>> =
+            Blake2bWrite::init(reader);
+        println!("[Ok] Read Transcript");
+
+        let vk: &VerifyingKey<G1Affine> = pk.get_vk();
+
+        let circuit_params = circuit.params();
+        //let vk_serialized = vk.to_bytes(halo2_base::halo2_proofs::SerdeFormat::RawBytes);
```

Comment:
what about this commented code?

Reply:
I guess PR is still WIP, lmk if so

Reply:
That was included by accident, I removed it now :).